pkg/mapper/fsm/fsm.go

```diff
@@ -248,7 +250,7 @@ func TestIfNeedBacktracking(mappings []string, orderingDisabled bool) bool {
 		metricRe = strings.Replace(metricRe, "*", "([^.]*)", -1)
 		regex, err := regexp.Compile("^" + metricRe + "$")
 		if err != nil {
-			log.Warnf("invalid match %s. cannot compile regex in mapping: %v", mapping, err)
+			level.Warn(logger).Log("msg", fmt.Sprintf("invalid match %s. cannot compile regex in mapping: %v", mapping, err))
```

Comment:
This should be converted to structured logging style.

```suggestion
			level.Warn(logger).Log("msg", "Invalid match, cannot compile regex in mapping", "mapping", mapping, "err" err))
```